apps/sverigesradio/sveriges_radio.star

```diff
@@ -0,0 +1,187 @@
+""")
+
+P4_ICON = base64.decode("""
+iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAYAAAA7MK6iAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAPpSURBVHgBvVZNbFtFEP52/eKYFKgJVCoosZxGIBpFYBR6qJSa5kbFoYFD01CJuDfEXxIOSAghmgMHOEAiekBCog4Sos6F9MKFgmmKqFQkMD9tVRK1r880CgUq9yepEz/vdva5fs7GjvMcOf0kW29nZ+fb2ZmdHQYP+DX8dfAm7AgTYq8EdgMsDMhgYZZlwJBiUqaExIloum/Si022JqGwB4lkqES0pkFTgscNbo/vNPvNKnqVcTI0QYQ45JWw0gaEZCPR9L74KvPlmAolPqaJIdQBZGe02+obriBfSTpxhEHGUEcwsHi3te/gchnXSZWn9SVVkGTzR7INbTNF0taJGGPyCDYQtIHhqLV/1CU+Ff4qnBc8SVcl7AgDPkfNhX3n2yhFRmbF6gwNdLi+FVEkG9KWGZv723rM5zOGIxPGACDCRZ0dp56Dvzngrjn/5mn8d+wS2t5+Ao/EHnNkucwS5o7O4NIHf9LSkn3W5EPXt3sQaGlyZYv/3sLvvd9j8e/5oCGWVNIe4gW3RQxV4ewW1kdnkF+wC04F/Wh9uQNt70Q0zdbXtmukCunD5xRpcTiYpPrAp1oTvcUjXgv2tRzmp69psodfanczJdB+L0KvdmjzWWsec1/MLBcFDWQjnBJqN2pA7kpWG3O/D0Zzo/P96Ic7yvT/eDGphcKBYL0cjD+JGlCWNMpONo+tB7Zh89NbNPnlz//CYnqh3AZjz6gYR1ADfMEGbZyl2DXc70fLK9vL5OlPzlW0ISXCnP491+JGSprNTz3kjkVOYObdX9Dy+uOUUJs0XWv0DBoebMSmzgfAneupIWjAI/g9PoTe6oR9w6YkW8LV5CzmvrwAO2uj88AuTXfumOnEtev4s8o//Nz9DR35vKZj0IFnvHgtbuUx/cZpTC+TSQp313d7dD2K9+yn59HxWXc1cxl11CbWia3929DUfp8msw6fxZa9obKj18BYio5anqCvmhKsCHshh4vvp9yxyEn8c/QCdp59ofpCKX4zwOUk3atBrAP/T6bLZEazn24oq87L2Q/cRiDlxPkugTGYUbNv0lAvxclQYoxk7xUnMz9dgREoJLx6lxYvL3g2LPMCV4/Plsb0s68vlcZSjDsbUH+qaBsyd7GWO70eKG9zTPT0UBPovE7Ka9rJCDYYgomRnjudp9v6FDoDOYYNgxyLmv3x4qhCs5eIk3AA9aSEGI9a/bHlMr5SKWr1xerrOXm6grQiscIua/8QxfygSgasF3RFqWMdVrYqTldbm6Qm0BCg7pMPqKcMHgnpTo3ZPDCqknZVNXjEVDjRywR1K4XGIeJePaf4UL2XVHqpCqqCVI2wiNt1wYXcpK13tgAAAABJRU5ErkJggg==
+""")
+
+def get_current_unix_time():
+    return time.now().unix * 1000
+
+def get_current_show_name(url):
+    response = fetch_data(url)
+    schedule_data = response["schedule"]
+    current_time = get_current_unix_time()
+
+    for episode in schedule_data:
+        start_time = int(episode["starttimeutc"].replace("/Date(", "").replace(")/", ""))
+        end_time = int(episode["endtimeutc"].replace("/Date(", "").replace(")/", ""))
+        if start_time <= current_time and current_time <= end_time:
+            return episode
+
+    return None
+
+def get_station_icon(station):
+    if station == "132":
+        return P1_ICON
+    elif station == "163":
+        return P2_ICON
+    elif station == "164":
+        return P3_ICON
+    else:
+        return P4_ICON
+
+def fetch_data(url):
```

Comment:
Note that the http module now has caching built in. Just pass `ttl_seconds = 10` to `http.get()` and you'll be able to drop most/all of this logic. Also future proofs against potential scope changes to the regular cache module down the line.